Add documentation to EAM's User and Tech guides

crterai opened the pull request:
This PR adds documentation related to the EAM, with specific emphasis on the schemes and input datasets used in EAMv3. 

___________________________________________________________
Contributions from @susburrows, Jack Chen, @brhillman, @DJFJJA, Guang Zhang, @hwangacme, @mingxuanwupnnl, @wlin7, @vlarson, @Jiwengithub, @jinboxie, @tangq, @yfenganl, Manish Shrivastava, @mt5555, @kaizhangpnl, @zyuying, @chengzhuzhang, Cheng Tao, @keziming

keziming commented on components/eam/docs/tech-guide/mam5.md:
```diff
@@ -0,0 +1,21 @@
+# Five-mode Modal Aerosol Model
```
@keziming @crterai I just updated this part.

tangq commented on components/eam/docs/tech-guide/chemUCIlinozv3.md:
```diff
@@ -0,0 +1,28 @@
+| ---------------------------- | ------------------------------------------------------------------------ | ---------------------- |
+| `airpl_emis_file`            | Aviation emission                                                        |                        |
+| `chlorine_loading_file`      | Chlorine loading                                                         |                        |
+| `chlorine_loading_fixed_ymd` | Directory of P3 look-up tables                                           |                        |
+| `chlorine_loading_type`      | P3 look-up table Version                                                 |                        |
```
The descriptions are for P3, not for these chlorine-related settings. The variable names should be self-explanatory because of the chlorine loading description above. It seems reasonable to leave these blanks empty.

tangq replied:
```suggestion
| `chlorine_loading_fixed_ymd` |                                            |                        |
| `chlorine_loading_type`      |                                                  |                        |
```

tangq commented on components/eam/docs/tech-guide/chemUCIlinozv3.md:
```diff
@@ -0,0 +1,28 @@
+| `chlorine_loading_fixed_ymd` | Directory of P3 look-up tables                                           |                        |
+| `chlorine_loading_type`      | P3 look-up table Version                                                 |                        |
+| `ext_frc_specifier`          | 3-D emissions                                                            |                        |
+| `ext_frc_cycle_yr`           | Output of P3 microphysical process rates                                 |                        |
+| `ext_frc_type`               | Tunable parameter for adjusting rain accretion efficiency                |                        |
```
Mismatched descriptions. Delete the descriptions.

tangq replied:
```suggestion
| `ext_frc_cycle_yr`           |                                  |                        |
| `ext_frc_type`               |                 |                        |
```

tangq commented on components/eam/docs/tech-guide/chemUCIlinozv3.md:
```diff
@@ -0,0 +1,28 @@
+| `ext_frc_cycle_yr`           | Output of P3 microphysical process rates                                 |                        |
+| `ext_frc_type`               | Tunable parameter for adjusting rain accretion efficiency                |                        |
+| `srf_emis_specifier`         | Surface emissions                                                        |                        |
+| `srf_emis_cycle_yr`          | Radius of embryomic raindrops from auto-conversion                       |                        |
+| `srf_emis_type`              | Upper bound of mean raindrop diameter                                    |                        |
```
Again, delete these descriptions.

tangq replied:
```suggestion
| `srf_emis_cycle_yr`          |                        |                        |
| `srf_emis_type`              |                                     |                        |
```

tangq commented on components/eam/docs/tech-guide/chemUCIlinozv3.md:
```diff
@@ -0,0 +1,28 @@
+| `srf_emis_cycle_yr`          | Radius of embryomic raindrops from auto-conversion                       |                        |
+| `srf_emis_type`              | Upper bound of mean raindrop diameter                                    |                        |
+| `linoz_data_file`            | Linoz data file                                                          |                        |
+| `linoz_data_cycle_yr`        | Nc exponent in droplet auto-conversion                                   |                        |
+| `linoz_data_path`            | Qc exponent in rain accretion                                            |                        |
+| `linoz_data_type`            | Qc exponeent in droplet autoconversion                                   |                        |
```
Delete these descriptions too.

tangq replied:
```suggestion
| `linoz_data_type`            |                                    |                        |
```

tangq reviewed (changes requested):
Hi @crterai , thanks for drafting this PR. I noticed some mismatched descriptions of input variables.

tangq commented on components/eam/docs/tech-guide/index.md:
```diff
@@ -1 +1,45 @@
+
+## Chemistry
+
+- [chemUCI + Linoz v3](chemUCIlinozv3.md): The parameterization of interactive atmospheric chemistry in EAMv3.
```
```suggestion
- [chemUCI + Linoz v3](chemUCIlinozv3.md): The interactive atmospheric chemistry packages in EAMv3.
```

tangq commented on components/eam/docs/tech-guide/chemUCIlinozv3.md:
```diff
@@ -0,0 +1,28 @@
+| `srf_emis_cycle_yr`          | Radius of embryomic raindrops from auto-conversion                       |                        |
+| `srf_emis_type`              | Upper bound of mean raindrop diameter                                    |                        |
+| `linoz_data_file`            | Linoz data file                                                          |                        |
+| `linoz_data_cycle_yr`        | Nc exponent in droplet auto-conversion                                   |                        |
+| `linoz_data_path`            | Qc exponent in rain accretion                                            |                        |
```
```suggestion
| `linoz_data_cycle_yr`        |                                   |                        |
| `linoz_data_path`            |                                             |                        |
```

tangq reviewed (changes requested):
Hi @crterai , thanks for drafting this PR. I noticed some mismatched descriptions of input variables. Since these variables are self-explanatory, they can be left as blanks.

rljacob commented on components/eam/docs/user-guide/index.md:
```diff
@@ -1 +1,139 @@
+
+### Namelist changes
+
+Namelist parameters can be specified in the `user_nl_eam` file. While most can be added at run time, some need to be added before `./case.setup` to have the changes applied in the simulation. 
```
What paramateres in user_nl_eam need to be specified before case.setup?

crterai replied:
Talking to @wlin7, they appear to be specifically in the chemistry and RRTMG, so I've rephrased this and will note them alongside the specific parameters.

rljacob replied:
@wlin7 I still don't understand how that works. You're saying that if I change something in user_nl_eam and then run "case.build", different code will be compiled?   I'm pretty sure CIME doesn't work that way.   The user namelist is strictly for run-time changes.

rljacob commented on components/eam/docs/user-guide/index.md:
```diff
@@ -1 +1,139 @@
+
+Namelist parameters can be specified in the `user_nl_eam` file. While most can be added at run time, some need to be added before `./case.setup` to have the changes applied in the simulation. 
+
+Refer to the individual schemes in the [tech-guide](../tech-guide/index.md) for a list of namelist parameters associated with each scheme in the atmosphere.
```
I think all the namelist paramaters should be covered in the user's guide since they are fundamental to **using** the model.

crterai replied:
That's a good point. Under the User's Guide section, I've created a new markdown document with all of the namelist parameters and linked it in the top level User's guide.

rljacob replied:
I like what I see in https://docs.e3sm.org/E3SM/pr-preview/pr-6314/EAM/user-guide/namelist_parameters/ but why doesn't that appear in the table of contents for the users-guide?

wlin7 commented on components/eam/docs/tech-guide/armdiags.md:
```diff
@@ -0,0 +1,47 @@
+## Overview
+
+The ARM data-oriented metrics and diagnostics package (ARM Diags) was developed to facilitate the use of ARM data in climate model evaluation and model intercomparison (Zhang et al. 2020). It includes ARM data sets, compiled from multiple ARM data products, and a Python-based analysis toolkit for computation ad visualization. It also includes simulation data from models participating the CMIP, wh ...
+
```
A typo above in 'Zhabg et al. 2022'

chengzhuzhang commented:
@rljacob we discussed about version control yesterday. Take this PR for example, where should we note that this doc is made for version3? Should we have EAMv3 (instead of EAM) under components?

rljacob commented:
All documentation is "for master" until we have complete documentation.

We decided long ago that the way to get older versions is on a maint branch.

bishtgautam commented on components/eam/docs/user-guide/index.md:
````diff
@@ -1 +1,139 @@
+
+#### Example output specification:
+
+```
````
This code block didn't get rendered correctly at https://e3sm-project.github.io/E3SM/pr-preview/pr-6314/EAM/user-guide/#example-output-specification

chengzhuzhang commented:
> All documentation is "for master" until we have complete documentation.
> 
> We decided long ago that the way to get older versions is on a maint branch.

Sounds good. Then maybe just leave a few lines in the summary of the main markdowns would be good.

crterai commented on components/eam/docs/tech-guide/index.md:
```diff
@@ -1 +1,45 @@
+
+## Aerosol
+
+- [MAM4](mam4.md): The primary parameterization scheme of aerosols in EAMv3.  
```
Thanks @keziming. I had intended MAM5 to describe the addition of the fifth mode for the stratospheric coarse mode and the MAM4 to describe all of the aerosol processes that we have inherited from EAMv0. Since MAM4 that we inherited covers most of the processes that we run in EAMv3, I had noted it as the primary parameterization scheme. 
Maybe there is a better way to describe what is under MAM4 and MAM5. @hwangacme - do you have any thoughts?

hwangacme replied:
Thanks @keziming for pointing this out. As I discussed with @crterai offline, a higher-level, explicit description of MAM5 as the default option of EAMv3 aerosol module is more preferable.

crterai replied:
Thanks @hwangacme. I'll make that change.

crterai commented on components/eam/docs/user-guide/index.md:
```diff
@@ -1 +1,139 @@
+
+#### Land-use / land cover change
+
+- <span style="color:red">Info needed on land-use land cover change / land surface data</span>
```
@bishtgautam - Could I get some help with a short description of the land-use land-cover change that we would use in the F2010, F1850, and F20TR simulations?

crterai commented on components/eam/docs/user-guide/aerosol_phys_prop.md:
```diff
@@ -0,0 +1,109 @@
+
+- BC and POM hygroscopicity values are updated by Shan et al. (2024).
+
+- <span style="color:red">Some description/reference about the 5th mode.</span>
```
@keziming - would you suggest a short description of the 5th mode here?

keziming replied:
Physical properties of the fifth mode aerosols are documented by Ke et al. (2024, in preparation)

keziming commented on components/eam/docs/user-guide/aerosol_phys_prop.md:
```diff
@@ -0,0 +1,111 @@
+- Original physical properties of aerosols are documented in Liu et al. (2012) [@liu_toward_2012]. Detailed information is included in the supplement.
+
+- Physical properties of aerosols used in E3SMv1 are documented in Wang et al. (2020) [@wang_aerosols_2020].
+
```
Physical properties of the fifth mode aerosols are documented by Ke et al. (2024, in preparation)

crterai commented:
@tangq @rljacob - I've updated the PR to address your suggested changes. Would you please review the changes?

crterai commented:
Removed WIP label, since I believe all reviewer suggestions have been addressed.

crterai commented:
Pinging @susburrows and @chengzhuzhang since I believe you'd also want to review this before it is merged.

rljacob commented on components/eam/docs/tech-guide/armdiags.md:
````diff
@@ -2,20 +2,20 @@
 
-```
+```text
 fincl7 = 'PS','Q','T','Z3','CLOUD','CONCLD','CLDICE','CLDLIQ','FREQR','REI','REL','PRECT','TMQ','PRECC','TREFHT','QREFHT','OMEGA','CLDTOT','LHFLX','SHFLX','FLDS','FSDS','FLNS','FSNS','FLNSC','FSDSC','FSNSC','AODVIS','AODABS','LS_FLXPRC','LS_FLXSNW','LS_REFFRAIN','ZMFLXPRC','ZMFLXSNW','CCN1','CCN2','CCN3','CCN4','CCN5','num_a1','num_a2','num_a3','num_a4','so4_a1','so4_a2','so4_a3','AREL','TGCLDLWP ...
````
Please break this line up.  Its not rendering correctly.

rljacob commented:
This some great content!

I don't see the references rendering correctly in the Tech Guide.
I think you should drop the subheadings in the Tech Guide main page.  There's not enough content under each subheading (like "Microphysics) to justify it.

rljacob commented:
Since MAM5 is the scheme used, don't have a separate section for MAM4.  Just combine that text in the explanation of MAM5.

rljacob commented:
Please rebase this to master so the linter can run on it.

hwangacme commented:
> Since MAM5 is the scheme used, don't have a separate section for MAM4. Just combine that text in the explanation of MAM5.

I am a little confused here. Is the User and Tech guides for EAMv3 only? MAM4 is the default scheme for v1 and v2. Also, MAM5 is based on MAM4 and there is no published documentation of MAM5 yet.

rljacob commented:
> I am a little confused here. Is the User and Tech guides for EAMv3 only? MAM4 is the default scheme for v1 and v2. Also, MAM5 is based on MAM4 and there is no published documentation of MAM5 yet.

The guides are for current master.  If MAM4 is not an option in current master, then don't give it its own section.  If you need to explain MAM4 in order to explain MAM5, do that in a single MAM5 section.

crterai commented:
> > I am a little confused here. Is the User and Tech guides for EAMv3 only? MAM4 is the default scheme for v1 and v2. Also, MAM5 is based on MAM4 and there is no published documentation of MAM5 yet.
> 
> The guides are for current master. If MAM4 is not an option in current master, then don't give it its own section. If you need to explain MAM4 in order to explain MAM5, do that in a single MAM5 section.

Does a plan to combine both under an umbrella MAM section satisfy both your concerns? @rljacob @hwangacme Or does it just confuse things more?
On the top Tech-docs page, I'll add the following description:
```
MAM (link): The parameterization schemes used to represent aerosols in EAMv3, consisting of the fifth mode in the stratospheric aerosols (MAM5), which was added on top of existing MAM4 schemes in EAMv1 and v2. 
```

rljacob commented:
Yes that's a good plan.

crterai commented:
> I think you should drop the subheadings in the Tech Guide main page. There's not enough content under each subheading (like "Microphysics) to justify it.

Yeah, I agree since the same information shows up in the short description of each parameterization. 
I'll create two subheadings: one for Dynamics / Physics and another for Diagnostic outputs.

crterai commented:
> @rljacob @crterai: I fixed the bib linking (was broken before my commit) and rebased to bring in all features from the main branch.
> 
> In my mind there are two items worth improving (but could be delayed to a separate PR):
> 
>     1. the nav bar should include all these added pages (e.g., see the soft warning below). We can do this either manually or more automatically later (I am working on a general solution)
> 
>     2. There are fortran code blocks in there (the namelist ones are actually fortran, not text, but that's not important)
> 
> 
> ```
> INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
>              - EAM/tech-guide/armdiags.md
>              - EAM/tech-guide/chemUCIlinozv3.md
>              - EAM/tech-guide/clubb.md
>              - EAM/tech-guide/cosp.md
>              - EAM/tech-guide/dust.md
>              - EAM/tech-guide/homme.md
>              - EAM/tech-guide/mam4.md
>              - EAM/tech-guide/mam5.md
>              - EAM/tech-guide/oceanfilms.md
>              - EAM/tech-guide/p3.md
>              - EAM/tech-guide/rrtmg.md
>              - EAM/tech-guide/vbs.md
>              - EAM/tech-guide/zm.md
>              - EAM/user-guide/aerosol_phys_prop.md
>              - EAM/user-guide/emission_oxidant_files.md
>              - EAM/user-guide/namelist_parameters.md
>              - EAMxx/technical/index.md
>              - ELM/dev-guide/testing.md
>              - ELM/tech-guide/top_solar_parameterization.md
> ```

Thanks for rebasing and bringing in the fixes, @mahf708. It'd be easy for me to fix the text to fortran. I don't know where to make those changes to the navbar though.

rljacob commented on components/eam/docs/index.md:
```diff
@@ -2,6 +2,6 @@
 
 Some introductory text here
```
This should be replaced with actual introductory text.  Something like "EAM is a state-of-the-art general circulation model of the atmosphere that ...."

crterai replied:
Thanks for catching this @rljacob. After consulting with Shaocheng, I've added a couple sentences with a quick overview of EAM.

rljacob commented on components/eam/docs/tech-guide/index.md:
```diff
@@ -1 +1,31 @@
+
+## Dynamics and Physics
+
+- [HOMME](homme.md): The dynamical core used in EAMv3.
```
"in EAMv3" everywhere is redundant.  I'll fix it.

rljacob commented on components/eam/docs/tech-guide/homme.md:
```diff
@@ -0,0 +1,26 @@
+EAM using the a dynamical core (dycore)  from the High Order Method Modeling Environment (HOMME).  The EAM dycore solves the atmospheric primitive equations governing the evolution of velocity, density, pressure and temperature, as well as the transport of water species and related hydrometers, aerosols and other atmospheric constituents.  The governing equations are written in a vertically lagra ...
+
+### References
+
```
Don't break out the references separately.  Keep them inline as done in other sections like P3 and MAM.  The bib function makes a better version of this.

crterai replied:
I've added the references in the overview text as suggested.

rljacob commented on components/eam/docs/tech-guide/clubb.md:
```diff
@@ -0,0 +1,23 @@
+
+### References
+
+- Larson (2022) [@larson_clubb-silhs_2022]
```
Don't break out the references separately. Keep them inline as done in other sections like P3 and MAM.  You can put all of these after the first sentence.

rljacob commented on components/eam/docs/tech-guide/cosp.md:
```diff
@@ -0,0 +1,36 @@
+
+### References
+
+- Zhang et al. (2024) [@zhang_understanding_2024]
```
Don't break out the references separately. Keep the footnote references inline as done in other sections like P3 and MAM.

rljacob commented on components/eam/docs/user-guide/namelist_parameters.md:
I'm concerned that the material here is duplicated in each of the markdown files in the tech-guide.  It should live in one place and I think it should be here.   You could link to each section here from the tech guide.  For example, in clubb.md, add 
```
 [Namelist Paramters](../user-guide/namelist_parameters.md#cloud-layers-unified-by-binormals)
```

crterai replied:
It makes sense to just have one copy so that any subsequent changes don't need to be made in two places. I've added the link, as you'd suggested.

rljacob commented:
I pushed some changes that mostly improve navigation.  This is almost ready to merge but I'd like to see the duplicate entries of namelist params removed and the references inlined in 3 of the tech guide pieces.

wlin7 commented on components/eam/docs/user-guide/index.md:
```diff
@@ -1 +1,175 @@
+
+`mfilt` - List that sets the number of timesteps to write in a single file before starting a new file.
+
+`avgflag_pertape` - List that sets the type of output to write. Choices are `'A'` for time-averaged output, `'A'` for instantaneous output, `'MIN'` for time-minimum output, and `'MAX'` for time-maximum output.
```
Need to replace `MIN` with `M`, and `MAX` with `X` for the avgflag above.

crterai replied:
Done.

crterai commented:
> I pushed some changes that mostly improve navigation. This is almost ready to merge but I'd like to see the duplicate entries of namelist params removed and the references inlined in 3 of the tech guide pieces.
I've pushed some changes that remove the duplicate entries of namelist parameters, add the references inline, add the top-level EAM description, and fixes the issue @wlin7 caught above.

crterai commented:
Thanks, @rljacob, @wlin7, and @mahf708 for improving and pushing this PR along.
